### PR TITLE
tooling: Use upstream distro packages

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: tooling: Use upstream distrotest
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/base/envoy_python.bzl
Comment (phlax) [inline tools/base/envoy_python.bzl:75]: @htuch im kinda surprised there isnt anything in `rules_ptyhon` like this to automatically make a command from a python entrypoint/function etc

either way, i think this is pretty useful and removes a load of boilerplate

File: tools/distribution/BUILD
Comment (phlax) [inline tools/distribution/BUILD:15]: im wondering if `entry_point` is a better name for this, i didnt call it that mostly because entry_points arent just for `console_scripts` - maybe we should call it that

Comment (htuch) [reply tools/distribution/BUILD:15]: Yes, please call it `entry_point`. And also rename to `envoy_py_binary` with this new support.

**Review by htuch — COMMENTED** [review]
LGTM, I'd recommend maybe doing a PR to rules_python to make this a generic supported thing as well? We can land this in Envoy for now with a TODO to cleanup.

**Review by htuch — COMMENTED** [review]
LGTM once we merge the script PR (just had a documentation nit there and ready to rubber stamp).

**Review by htuch — APPROVED** [review]

**Pull request merged**
